core/src/main/scala/kafka/log/Log.scala

```diff
@@ -100,7 +102,8 @@ case class LogAppendInfo(var firstOffset: Option[Long],
                          offsetsMonotonic: Boolean,
                          lastOffsetOfFirstBatch: Long,
                          recordErrors: Seq[RecordError] = List(),
-                         errorMessage: String = null) {
+                         errorMessage: String = null,
+                         leaderHWIncremented: Option[Boolean] = None) {
```

Comment:
Can we not use a `boolean` here? `false` until it's been incremented and then `true`. Is there value in having the third state?